### PR TITLE
Introduce new methods on `RequestContext`

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

> $Subject

Fixes : https://github.com/ballerina-platform/ballerina-standard-library/issues/3090
Fixes : https://github.com/ballerina-platform/ballerina-standard-library/issues/4070

## Examples

```bal
public function main() returns error? {
    http:RequestContext ctx = new();
    ctx.set("key", "value");
    
    if ctx.hasKey("key") {
        string value = check ctx.get("key").ensureType();
        io:println("Context value : " + value.toString());
    }

    io:println("Context keys : " + ctx.keys().toString());

    string value = check ctx.getWithType("key");
    io:println("Context value : " + value.toString());
}
```

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec
- [x] [Checked native-image compatibility](https://github.com/ballerina-platform/module-ballerina-http/actions/runs/4160079965)

### CODE REVIEW

File: ballerina-tests/http-interceptor-tests/tests/req_ctx_methods_test.bal [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import ballerina/test;
```
Comment (chamil321) [inline ballerina-tests/http-interceptor-tests/tests/req_ctx_methods_test.bal:1]: Licence header

File: native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternRequestContext.java [hunk 2]
```diff
@@ -18,15 +18,30 @@
...
  */
 public class ExternRequestContext {
+
+    public static Object getWithType(BObject requestCtx, BString key, BTypedesc targetType) {
+        BMap attributes = requestCtx.getMapValue(HttpConstants.REQUEST_CTX_ATTRIBUTES);
+        try {
+            Object value = attributes.getOrThrow(key);
+            return EnsureType.ensureType(value, targetType);
+        } catch (Exception exp) {
+            return exp;
```
Comment (chamil321) [inline native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternRequestContext.java:41]: Create the `ListenerError` and return

File: ballerina-tests/http-interceptor-tests/tests/req_ctx_methods_test.bal (hunk 1 above)
Comment (TharmiganK) [reply ballerina-tests/http-interceptor-tests/tests/req_ctx_methods_test.bal:17]: Fixed via [6f1e08c](https://github.com/ballerina-platform/module-ballerina-http/pull/1438/commits/6f1e08c53056c40a1aa9d3440d2468fa03c9388c)

File: native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternRequestContext.java (hunk 2 above)
Comment (TharmiganK) [reply native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternRequestContext.java]: Fixed via [d4c8dff](https://github.com/ballerina-platform/module-ballerina-http/pull/1438/commits/d4c8dff8715856861a7af52175afc813f729fd4a)

**Comment by shafreenAnfar** [comment]
What happens if don't have the `key`

```
string value = check ctx.getWithType("key");
```

**Comment by TharmiganK** [comment]
> What happens if don't have the `key`
> 
> ```
> string value = check ctx.getWithType("key");
> ```

An error will be returned : 

> What happens if don't have the `key`
> 
> ```
> string value = check ctx.getWithType("key");
> ```

It will return an error rather than panicking. The error would be `no attribute found for key: key`

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**